backend/app/database/user.py

```diff
@@ -0,0 +1,51 @@
+#  Copyright (c) 2020-2023. KennelTeam.
+#  All rights reserved.
+from backend.config_loader import ConfigLoader
+from . import db
+from .editable_mixin import EditableMixin
+from sqlalchemy.dialects.mysql import VARCHAR
+import random
+import string
+import hashlib
+from enum import Enum
+
+
+class Role(Enum):
+    UNAUTHORIZED = 0
+    GUEST = 1
+    INTERN = 2
+    EDITOR = 3
+    ADMIN = 4
+
+
+class User(EditableMixin, db.Model):
+    __tablename__ = 'users'
+    login = db.Column('login', VARCHAR(ConfigLoader.get_config("MAX_LOGIN_SIZE")), unique=True)
+    name = db.Column('name', db.Text(ConfigLoader.get_config("MAX_FULLNAME_SIZE")))
+    comment = db.Column('comment', db.Text(ConfigLoader.get_config("MAX_COMMENT_SIZE")))
+
+    # expected to use SHA-512
+    password_hash = db.Column('password_hash', db.Text(512 // 8))
+
+    # the length of salt is just made up
+    password_salt = db.Column('password_salt', db.Text(ConfigLoader.get_config("SALT_SIZE")))
+    role = db.Column('role', db.Integer)
+
+    @staticmethod
+    def __generate_salt():
```

Comment:
Why double underscore instead of just a single underscore? Double underscore results in name mangling (the attribute becomes `User__generate_salt` instead of pure `__generate_salt`) and I don't think that it is needed here.